lib/chef/user.rb

```diff
@@ -144,12 +287,17 @@ class << self
     end
 
     def self.list(inflate=false)
-      response = Chef::REST.new(Chef::Config[:chef_server_url]).get_rest('users')
-      users = if response.is_a?(Array)
```

Comment:
I'll probably end up emailing this PR to dev or the client team or something. Definitely want a lot of review on this. Would like to nail this API versioning stuff the first time.